### ISSUE TITLE
Cannot execute blocks on sender in exclusive mode

### ISSUE THREAD

**Issue by kostya** [celluloid/celluloid#327]
Gemfile:
```
source :rubygems
gem 'celluloid', '~> 0.15.0'
gem 'celluloid-io', '~> 0.15.0'
gem 'reel', :git => "https://github.com/celluloid/reel.git"
```

1.rb
```ruby
require 'bundler/setup'
require 'reel'
require 'celluloid'

class A
  include Celluloid

  exclusive :start_server

  def run
    start_server
  end

  def start_server
    @server = Reel::Server.supervise('127.0.0.1', 12345) do |connection|
      while request = connection.request
        connection.respond(Reel::Response.new(:ok, "Hello world"))
      end
    end
  end
end

a = A.new
a.start_server
```

```
ruby 1.rb 
D, [2013-09-06T21:54:28.265339 #2289] DEBUG -- : Terminating 5 actors...
E, [2013-09-06T21:54:28.266123 #2289] ERROR -- : A crashed!
RuntimeError: Cannot execute blocks on sender in exclusive mode
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:11:in `initialize'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:58:in `initialize'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/proxies/sync_proxy.rb:31:in `new'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/proxies/sync_proxy.rb:31:in `method_missing'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/proxies/actor_proxy.rb:20:in `_send_'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid.rb:185:in `new'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/supervisor.rb:10:in `supervise'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid.rb:204:in `supervise'
	1.rb:15:in `start_server'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:25:in `public_send'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:25:in `dispatch'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:67:in `dispatch'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/actor.rb:322:in `block in handle_message'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/actor.rb:414:in `block (2 levels) in task'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/tasks.rb:109:in `exclusive'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid.rb:427:in `exclusive'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/actor.rb:414:in `block in task'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/tasks.rb:55:in `block in initialize'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/tasks/task_fiber.rb:13:in `block in create'
	(celluloid):0:in `remote procedure call'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:92:in `value'
	/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/proxies/sync_proxy.rb:33:in `method_missing'
	1.rb:24:in `<main>'
/home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:11:in `initialize': Cannot execute blocks on sender in exclusive mode (RuntimeError)
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:58:in `initialize'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/proxies/sync_proxy.rb:31:in `new'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/proxies/sync_proxy.rb:31:in `method_missing'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/proxies/actor_proxy.rb:20:in `_send_'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid.rb:185:in `new'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/supervisor.rb:10:in `supervise'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid.rb:204:in `supervise'
	from 1.rb:15:in `start_server'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:25:in `public_send'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:25:in `dispatch'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:67:in `dispatch'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/actor.rb:322:in `block in handle_message'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/actor.rb:414:in `block (2 levels) in task'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/tasks.rb:109:in `exclusive'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid.rb:427:in `exclusive'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/actor.rb:414:in `block in task'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/tasks.rb:55:in `block in initialize'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/tasks/task_fiber.rb:13:in `block in create'
	from (celluloid):0:in `remote procedure call'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/calls.rb:92:in `value'
	from /home/kostya/.rvm/gems/ruby-2.0.0-p0/gems/celluloid-0.15.1/lib/celluloid/proxies/sync_proxy.rb:33:in `method_missing'
	from 1.rb:24:in `<main>'
```

**Comment by digitalextremist** [comment]
Is this still relevant? It has become very old.

**Comment by digitalextremist** [comment]
If this is not resolved in #511 please reopen.
